schema/commands.proto

```diff
@@ -1,293 +1,69 @@
-  }
-
-  // transaction "creator" invites account with account_id to the domain
-  message InviteAccount {
-    string account_id = 1; // invitee
-    string domain = 2; // invite to this domain
-    uint32 permissions = 3; // logical or of Permissions enum
-    string username = 4; // account will join with this username: <username>@soramitsu.co.jp
+    string account_name = 1;
+    string asset_name = 2;
+    string amount = 3;
   }
 
-  // remove account from a domain (creator should have according permissions)
-  // this is for domain admins to kick accounts from a domain
-  message RemoveAccount {
-    string username = 1; // full username with full domain name: username@domain
+  message Create{
+    string asset_name = 1;
+    uint32 precision = 2;
   }
 
-  // set permissions for the account in domain
-  message SetPermissions {
-    string username = 1; // full username with full domain name: username@domain
-    uint32 permissions = 2;
-  }
 }
 
 
 message Account {
   // create new account with quorum and signatories
   // account id is defined as the first public key
   // account id = sigs[0].pubkey
-  message Create {
-    uint32 quorum = 1;
-    repeated Signature sigs = 2;
-  }
-
-  // account who created transaction ("creator") sets quorum
-  message SetQuorum {
-    int32 quorum = 1;
-  }
-
-  // account who created transaction ("creator") registers signatories
-  message RegisterSignatories {
-    repeated Signature sigs = 1;
-  }
-
-  // account who created transaction ("creator") derregisters signatories
-  message DeregisterSignatories {
-    repeated Signature sigs = 1;
-  }
-
-  enum State {
-    active = 0;
-    suspended = 1; // banned
-    // not_approved = 2; // whatever
-  }
-
-  // set account state
-  message SetState {
-    string account_id = 1;
-    State state = 2;
+  message AddSign {
+    string account_name = 1;
+    bytes pubkey = 2;
   }
 
-  // account who created transaction ("creator") joins a domain
-  message JoinDomain {
-    string domain = 1;
-    string username = 2;
-  }
-
-  // account who created transaction ("creator") leaves domain
-  message LeaveDomain {
-    string domain = 1;
+  message Create {
+    string account_name = 1;
+    bytes main_pubkey = 2;
   }
-}
-
 
-message Peer {
-  enum Role {
-    validator = 0; // now all peers are validators, but this is for future versions
-    member = 1;
+  message RemoveSign {
+    string account_name = 1;
+    bytes pubkey = 2;
   }
-
-  // add peer to the network
-  message Add {
-    string ip = 1;
-    string account_id = 2;
-    Role role = 3;
+  message SetPermissions {
+    string account_name = 1;
+    uint32 permissions = 2;
   }
 
-  // remove peer from the network
-  message Remove {
-    string ip = 1;
+  message SetQuorum {
+    string account_name = 1;
+    int32 quorum = 2;
   }
 
-  // set peer role in the network
-  message SetRole {
-    string ip = 1;
-    Role role = 2;
+  message Transfer {
```

Comment:
I think `Asset{}`, `Account{}`, `Domain{}`, and so on are related by Permission model.
So, `Asset` has permission that are `Add`, `Subtract`, `Transfer`.